src/concrete_solve.jl

```diff
@@ -420,9 +461,9 @@ function DiffEqBase._concrete_solve_adjoint(
         no_start && (sol_idxs = sol_idxs[2:end])
         no_end && (sol_idxs = sol_idxs[1:(end - 1)])
         only_end = length(sol_idxs) <= 1
-        _u = sol.u[sol_idxs]
-        u = save_idxs === nothing ? _u : [x[save_idxs] for x in _u]
-        ts = sol.t[sol_idxs]
+        uf = save_idxs === nothing ? getu(sol, :) : getu(sol, save_idxs)
```

Comment:
Is `getu` actually the same here? Won't that append observed? I don't know the `getu` `:` behavior but this seems like a weird change to include?